src/data/nat/prime.lean

```diff
@@ -593,6 +594,27 @@ lemma mem_factors {n p} (hn : 0 < n) : p ∈ factors n ↔ prime p ∧ p ∣ n :
 ⟨λ h, ⟨prime_of_mem_factors h, (mem_factors_iff_dvd hn $ prime_of_mem_factors h).mp h⟩,
  λ ⟨hprime, hdvd⟩, (mem_factors_iff_dvd hn hprime).mpr hdvd⟩
 
+lemma factors_subset {n k : ℕ} (h : k ≠ 0) : n.factors ⊆ (n * k).factors :=
+begin
+  cases n,
+  { rw zero_mul, refl },
+  cases n,
+  { rw factors_one, apply list.nil_subset },
+  intros p hp,
+  rw mem_factors at ⊢ hp,
+  exact ⟨hp.1, dvd_mul_of_dvd_left hp.2 k⟩,
+  exact nat.mul_pos succ_pos' (nat.pos_of_ne_zero h),
+  exact succ_pos'
+end
+
+lemma factors_subset_of_dvd {n k : ℕ} (h : n ∣ k) (h' : k ≠ 0) : n.factors ⊆ k.factors :=
```

Comment:
Same for `0 < k`

Reply:
hmm, do we really not have a `nat.right_pos_of_mul_pos`? I can't seem to find it, but I agree that probably we should switch to `lt` if we can

Reply:
There is `pos_of_mul_pos_right` for ordered rings, but that needs a nonneg assumption. As above, I'm happy to keep the `≠`.